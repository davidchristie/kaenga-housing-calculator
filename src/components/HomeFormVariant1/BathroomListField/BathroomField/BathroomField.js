import React, { Component, PropTypes } from 'react'
import { Button, ListGroupItem } from 'react-bootstrap'
import { Field } from 'redux-form'

import CheckboxControl from '../../../controls/CheckboxControl'

class BathroomField extends Component {
  render () {
    const { index, bathroom, remove } = this.props
    return (
      <ListGroupItem>
        <h4>Bathroom #{index + 1}</h4>
        <Field
          name={`${bathroom}.shower`}
          component={CheckboxControl}
        >
          Shower
        </Field>
        <Field
          name={`${bathroom}.bath`}
          component={CheckboxControl}
        >
          Bath
        </Field>
        <Button onClick={remove}>
          Remove Bathroom
        </Button>
      </ListGroupItem>
    )
  }
}

BathroomField.defaultProps = {
  bathroom: '',
  index: 0
}

BathroomField.propTypes = {
  bathroom: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  remove: PropTypes.func
}

export default BathroomField
