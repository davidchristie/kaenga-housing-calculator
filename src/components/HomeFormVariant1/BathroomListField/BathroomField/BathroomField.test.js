import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import BathroomField from './BathroomField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<BathroomField />)).toMatchSnapshot()
})
