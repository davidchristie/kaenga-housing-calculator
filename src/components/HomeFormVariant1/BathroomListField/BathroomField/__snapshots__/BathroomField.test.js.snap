exports[`test matches snapshot 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  listItem={false}>
  <h4>
    Bathroom #
    1
  </h4>
  <Field
    component={[Function]}
    name=".shower">
    Shower
  </Field>
  <Field
    component={[Function]}
    name=".bath">
    Bath
  </Field>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={undefined}>
    Remove Bathroom
  </Button>
</ListGroupItem>
`;
